hack/debug-network.sh

```diff
@@ -0,0 +1,574 @@
+#!/bin/bash
+
+# echoes args to stderr and exits
+die () {
+    echo "$*" 1>&2
+    exit 1
+}
+
+# echoes the command provided as $@ and then runs it
+echo_and_eval () {
```

Comment:
If a user wants debug output they can `set -x`, if we always want to print something, we should print it. I don't like messing with `eval` if we can avoid it, there are quite a lot of loopholes and gotchas.